MapboxVision/Services/BluetoothManager.swift

```diff
@@ -0,0 +1,37 @@
+import CoreBluetooth
+
+/// Class that contains logic to dispatch and manage the bluetooth module.
+final class BluetoothManager: NSObject {
+    // MARK: Public properties
+
+    /// Indicates whether the bluetooth module is powered on.
+    private(set) var isBluetoothPoweredOn = false
+
+    // MARK: Private properties
+
+    /// Underlying `CBCentralManager` object that manages state of bluetooth module.
+    private var coreBluetoothManager: CBCentralManager?
+    /// The dispatch queue to dispatch the CBCentralManager's events.
+    private let bluetoothManagerQueue = DispatchQueue(label: "com.mapbox.BluetoothManager")
+
+    // MARK: Lifecycle
+
+    /**
+     Base initializer to create an instance of `BluetoothManager`.
+     */
+    override init() {
+        super.init()
+        coreBluetoothManager = CBCentralManager(delegate: self, queue: bluetoothManagerQueue)
```

Comment:
The same questions about subscribing to updates. State is available on `CBManager.state`.

Reply:
The answer is similar to answer above. We don't need to pull the CentralManager's state each time (for now update interval in core equals to 60 seconds). We can use cached value from BluetoothManager instead and update it only when the bluetooth status is changed (there's a high probability it will be changing not so often).

Reply:
Agreed to get rid of subscriptions.